src/select-menu/src/OptionsList.js

```diff
@@ -189,14 +198,20 @@ export default class OptionsList extends PureComponent {
       nextIndex = 0
     }
 
+    if (this.isSelected(options[nextIndex])) {
+      return
+    }
+
     onSelect(options[nextIndex])
```

Comment:
Maybe instead of returning early we make the semantics clear by only calling `onSelect` when it's not selected already.

```js
if (!this.isSelected(options[nextIndex])) {
  onSelect(options[nextIndex])
}
```

Also, could you elaborate on when arrow down/up triggers selecting the item that's already selected?